### PR TITLE
Creates the starting point of our application and its architecture

### LINKED ISSUE #4: Write Seeds-function
The Seeds-function is the function that creates the sample data and sample users from scratch required for the basic application to run.

### LINKED ISSUE #7: Follow the Back-End tutorial: W-M
Follow the Back-End tutorial: 
https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-i-hello-world

at least up to and including 'Error Handling', to get a decent understanding of Flask.

### PR DESCRIPTION
**Pull request by Qqwy**
Adds:

- A simple Flask application using a couple of static pages.
- an architectural structure that we can continue to build up, hopefully structured enough for everyone to be clear 'where everything goes', so we can keep high-cohesion and low-coupling even later on in the project.
- Documentation of the different modules :wink: 

Functionality of application so far:

- User can log in (Everyone of the team (as well as Patrick) has an account which is their first name, properly capitalized. Their password is their first name, in lowercase.
- Attempting to log in with an unexistent username or wrong password will reject access.
- When logged in, users are redirected to a mostly-empty dashboard page.

Fixes #4 , fixes #7 .

### CODE REVIEW

File: pydash/config.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import os
+
```
Comment (FlyingBird95) [inline pydash/config.py]: PEP8: 2 blank lines

File: pydash/pydash_app/user/__init__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+"""
+This module is the public interface (available to the web-application pydash_web)
+for interacting with Users.
+"""
+from .user import User
+import pydash_app.user.user_repository
+
```
Comment (FlyingBird95) [inline pydash/pydash_app/user/__init__.py]: PEP8: 2 blank lines

File: pydash/pydash_app/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+"""
```
Comment (FlyingBird95) [inline pydash/pydash_app/__init__.py]: nice!

File: pydash/pydash_app/user/__init__.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+"""
```
Comment (FlyingBird95) [inline pydash/pydash_app/user/__init__.py]: nice!

File: pydash/pydash_app/user/user.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    - What properties a User has
+    - What functionality makes sense to have this User interact with information from elsewhere.
+
+    Per Domain Driven Design, it does _not_ contain information on how to persistently store/load a user!
+    (That is instead handled by the `user_repository`).
+    """
+    def __init__(self, name=None, password=None):
+        if name == None or password == None:
+            raise "Missing arguments to User constructor!"
```
Comment (FlyingBird95) [inline pydash/pydash_app/user/user.py]: maybe better: raise Exception("your string")

File: pydash/pydash_app/user/user.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+from werkzeug.security import generate_password_hash
+from werkzeug.security import check_password_hash
+from flask_login import UserMixin
+
```
Comment (FlyingBird95) [inline pydash/pydash_app/user/user.py]: PEP8: 2 blank lines

File: pydash/pydash_app/user/user.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+            raise "Missing arguments to User constructor!"
+
+        self.id = str(id(self))
+        self.name = name
+        self.password_hash = generate_password_hash(password)
+
+    def __repr__(self):
+        return '<{} {}>'.format(self.__class__.__name__, self.name)
+
+    # TODO Use an ID that is separate from the user's name.
```
Comment (FlyingBird95) [inline pydash/pydash_app/user/user.py]: nice!

File: pydash/pydash_app/user/user_repository.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
+"""
```
Comment (FlyingBird95) [inline pydash/pydash_app/user/user_repository.py]: perfect documentation!

File: pydash/pydash_web/__init__.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+from config import Config
+
+import pydash_app
+
+flask_webapp = Flask(__name__)
+flask_webapp.config.from_object(Config)
+login_manager = LoginManager(flask_webapp)
+login_manager.login_view = 'login'
+
+from pydash_web import routes  # Needs to be below flask_webapp instantiation to prevent circular dependency
```
Comment (FlyingBird95) [inline pydash/pydash_web/__init__.py]: ok

File: pydash/pydash_web/__init__.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+from config import Config
+
+import pydash_app
+
+flask_webapp = Flask(__name__)
+flask_webapp.config.from_object(Config)
+login_manager = LoginManager(flask_webapp)
+login_manager.login_view = 'login'
+
```
Comment (FlyingBird95) [inline pydash/pydash_web/__init__.py]: PEP8: 2 blank lines

File: pydash/pydash_web/controller/login.py [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+        return __render_login_form(login_form)
+
+    login_user(user, remember=login_form.remember_me.data)
+    flash("User {} successfully logged in!".format(user.name))
+    return redirect(__next_page())
+
+
+def __next_page():
+    next = request.args.get('next')
+    if not next or url_parse.netloc(next) != '':
```
Comment (FlyingBird95) [inline pydash/pydash_web/controller/login.py]: undefined url_parse

File: pydash/pydash_web/controller/logout.py [hunk 12]
```diff
@@ -0,0 +1,9 @@
+"""
+Allows a user to sign out again after finishing using the application
+"""
+from flask import redirect, url_for
+from flask_login import logout_user
+
```
Comment (FlyingBird95) [inline pydash/pydash_web/controller/logout.py]: PEP8: 2 blank lines

File: pydash/pydash_web/routes.py [hunk 13]
```diff
@@ -0,0 +1,28 @@
+"""
+Contains the different routes (web endpoints) that the pydash_web flask application can respond to.
+
+The actual implementation of each of the routes' dispatching logic is handled by the respective 'controller' function.
+"""
+
+from flask import render_template
```
Comment (FlyingBird95) [inline pydash/pydash_web/routes.py]: unused import

File: pydash/pydash_web/templates/base.html [hunk 14]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (FlyingBird95) [inline pydash/pydash_web/templates/base.html]: NICE!

File: pydash/config.py (hunk 1 above)
Comment (Qqwy) [reply pydash/config.py]: done

File: pydash/pydash_app/user/__init__.py (hunk 2 above)
Comment (Qqwy) [reply pydash/pydash_app/user/__init__.py]: done

File: pydash/pydash_app/user/user.py (hunk 6 above)
Comment (Qqwy) [reply pydash/pydash_app/user/user.py]: done

File: pydash/pydash_web/__init__.py (hunk 10 above)
Comment (Qqwy) [reply pydash/pydash_web/__init__.py]: done

File: pydash/pydash_web/controller/logout.py (hunk 12 above)
Comment (Qqwy) [reply pydash/pydash_web/controller/logout.py]: done

File: pydash/pydash_web/controller/login.py (hunk 11 above)
Comment (Qqwy) [reply pydash/pydash_web/controller/login.py]: Import has been added. I have no idea why even without the import, Flask was able to run this script without any problems. :man_shrugging:

**Pull request merged**